### PR TITLE
[Backport 2.x] Added org.apache.logging.log4j:log4j-slf4j-impl to classpath (#791)

### LINKED ISSUE #693: [BUG] Some slf4j warn message shows in the log file when sending email
**What is the bug?**
When sending a email at the first time, OpenSearch main log shows that there are some slf4j related warn message, I think it should be eliminated:
```
[2023-06-26T17:02:40,596][WARN ][stderr                   ] [bcd07466ee01] SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
[2023-06-26T17:02:40,597][WARN ][stderr                   ] [bcd07466ee01] SLF4J: Defaulting to no-operation (NOP) logger implementation
[2023-06-26T17:02:40,597][WARN ][stderr                   ] [bcd07466ee01] SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
```

**How can one reproduce the bug?**
1. Go to 'Alerting plugin of OSD '
2. Click on 'send test message' after you have setup email account correctly and creating a alerting monitor
3. See some warn log in OpenSearch main log file

**What is the expected behavior?**
Remove the unexpected warn log

**What is your host/environment?**
 - OS: [MacOS]
 - Version [3.0]
 - Plugins: notification, alerting

**Do you have any screenshots?**
image

### PR DESCRIPTION
**Pull request by Noir01**
* Added org.apache.logging.log4j:log4j-slf4j-impl to classpath

Signed-off-by: Aniruddh <63553175+Noir01@users.noreply.github.com>
(cherry picked from commit 62b7b4f840e7e224f947f1745da375df6f419b9b)

### Description
Fixes warning message shown when sending test message

### Issues Resolved
fixes #693

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by gaobinlong — APPROVED** [review]

File: notifications/core/build.gradle [hunk 1]
```diff
@@ -142,6 +142,8 @@ dependencies {
     implementation "com.amazonaws:aws-java-sdk-ses:${aws_version}"
     implementation "com.sun.mail:javax.mail:1.6.2"
     implementation "javax.activation:activation:1.1"
+    implementation "org.slf4j:slf4j-api:${versions.slf4j}" //Needed for httpclient5
+    implementation "org.apache.logging.log4j:log4j-slf4j-impl:${versions.log4j}"
     testImplementation(
```
Comment (Hailong-am) [inline notifications/core/build.gradle:147]: do we need backport 2.x? this is need for httpclient 5, 2.x branch still using 4.x

Comment (gaobinlong) [reply notifications/core/build.gradle:147]: By my test, the error log of log4j really not seen in 2.x version, will close this backport PR, thanks!

**Comment by gaobinlong** [comment]
Thanks @Noir01 , I will close this backport PR because it's not needed.

**Comment by Noir01** [comment]
Alrighty

**Pull request closed without merging**